File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.js
function solution(n) {
    const array1 = Array(n + 1).fill(true).fill(false, 0 , 2)
    for (let i = 2; i <= Math.sqrt(n); i++) {
        if (array1[i]) {
            for (let j = i * i; j <= n; j += i) {
                array1[j] = false
            }
        }
    }
    const answer = array1.filter(e=> e).length
    return answer;
}

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.py
import math
def solution():
    N = int(input())
    values = list(map(int, input().split()))
    count = N
    for value in values:
        if value == 1:
            count -= 1
            continue
        for i in range(2, int(math.sqrt(value)) + 1):
            if value % i == 0:
                count -= 1
                break
    return count
print(solution())